docs/src/main/asciidoc/security-oidc-code-flow-authentication-concept.adoc

```diff
@@ -283,269 +725,152 @@ quarkus.oidc.credentials.secret=secret
 
-=== Cross-origin resource sharing
+You can have this process further optimized by having a simple JavaScript function periodically emulating the user activity by pinging your Quarkus endpoint thus minimizing the window during which the user may have to be re-authenticated.
 
-If you plan to consume this application from a Single-page application running on a different domain, you will need to configure CORS (Cross-Origin Resource Sharing). For more information, see the xref:http-reference.adoc#cors-filter[HTTP CORS documentation].
+Note this user session can not be extended forever - the returning user with the expired ID token will have to re-authenticate at the OIDC provider endpoint once the refresh token has expired.
 
 [[oauth2]]
-=== Integration with GitHub and other OAuth2 providers
+=== Integration with GitHub and non-OIDC OAuth2 providers
 
 Some well known providers such as GitHub or LinkedIn are not OpenID Connect but OAuth2 providers which support the `authorization code flow`, for example, link:https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps[GitHub OAuth2] and link:https://docs.microsoft.com/en-us/linkedin/shared/authentication/authorization-code-flow[LinkedIn OAuth2].
+Remember, OIDC is built on top of OAuth2.
 
-The main difference between OpenID Connect and OAuth2 providers is that OpenID Connect providers, by building on top of OAuth2, return an `ID Token` representing a user authentication, in addition to the standard authorization code flow `access` and `refresh` tokens returned by `OAuth2` providers.
+The main difference between OpenID Connect and OAuth2 providers is that OpenID Connect providers return an `ID Token` representing a user authentication, in addition to the standard authorization code flow `access` and `refresh` tokens returned by `OAuth2` providers.
 
 OAuth2 providers such as GitHub do not return `IdToken`, the fact of the user authentication is implicit and is indirectly represented by the `access` token which represents an authenticated user authorizing the current Quarkus `web-app` application to access some data on behalf of the authenticated user.
 
-For example, when working with GitHub, the Quarkus endpoint can acquire an `access` token which will allow it to request a GitHub profile of the current user.
-In fact this is exactly how a standard OpenID Connect `UserInfo` acquisition also works - by authenticating into your OpenID Connect provider you also give a permission to Quarkus application to acquire your <<user-info,UserInfo>> on your behalf - and it also shows what is meant by OpenID Connect being built on top of OAuth2.
+While in the OIDC case, we validate the ID Token as proof of the authentication validity, in the raw OAuth2 case, we need to validate the access token.
+This is done by subsequently calling an endpoint that requires the access token and that typically returns user information.
```

Comment:
Question: What is meant by "subsequently" here? Also, is "typically" needed here? Are there instances where the user information is not returned? Could this sentence be adjusted as follows perhaps"
"You validate the access token by calling an endpoint that requires the token. 
This action returns the user information."

Reply:
Hi @sheilamjones It is probably the right balance there, as it tries to capture various variations without going into too many details. `subsequently` in this context means that after the code flow has completed, a follow up request is done to fetch more information, `typically` it would be a user information but in `OAuth2`, as opposed to `OIDC`, `UserInfo` is undefined, so whatever endpoint is available, like a user profile or something close enough that cab be used to indirectly validate the access token, will do...
So we should probably leave it as is